src/cpp/session/modules/build/SessionBuild.cpp

```diff
@@ -678,18 +678,20 @@ class Build : boost::noncopyable,
       type_ = type;
 
       // add testthat and shinytest result parsers
-      core::Version testthatVersion;
-      module_context::packageVersion("testthat", &testthatVersion);
-      
       if (type == kTestFile)
       {
          openErrorList_ = false;
-         parsers.add(testthatErrorParser(packagePath.getParent(), testthatVersion));
+         if (module_context::isPackageInstalled("testthat")) {
```

Comment:
Maybe we'd be better off with a method in `CompileErrorParsers`, i.e. `CompileErrorParsers::addTestthatErrorParser(<path>)` to avoid duplicating the `if` here.